protocols/frost/keygen/keygen_test.go

```diff
@@ -143,15 +143,49 @@ func checkOutputTaproot(t *testing.T, rounds []round.Session, parties party.IDSl
 
 func TestKeygenTaproot(t *testing.T) {
 	N := 5
+	rounds, partyIDs := taprootDkg(t, N)
+	checkOutputTaproot(t, rounds, partyIDs)
+}
+
+func TestRefreshTaproot(t *testing.T) {
+	N := 5
+	rounds, partyIDs := taprootDkg(t, N)
+
+	refreshRounds := make([]round.Session, 0, N)
+	for _, partyID := range partyIDs {
+		currentRound := rounds[len(refreshRounds)]
+		config := currentRound.(*round.Output).Result.(*TaprootConfig)
+		publicKey, err := curve.Secp256k1{}.LiftX(config.PublicKey)
+		require.NoError(t, err)
+		privateShare := config.PrivateShare
+		verificationShares := make(map[party.ID]curve.Point, len(config.VerificationShares))
+		for k, v := range config.VerificationShares {
+			verificationShares[k] = v
+		}
+
+		r, err := StartKeygenCommon(true, curve.Secp256k1{}, partyIDs, N-1, partyID, privateShare, publicKey, verificationShares)(nil)
+		require.NoError(t, err, "round creation should not result in an error")
+		refreshRounds = append(refreshRounds, r)
+	}
+
+	for {
+		err, done := test.Rounds(refreshRounds, nil)
+		require.NoError(t, err, "failed to process round")
+		if done {
+			break
+		}
+	}
+	checkOutputTaproot(t, refreshRounds, partyIDs)
+}
+
+func taprootDkg(t *testing.T, N int) ([]round.Session, party.IDSlice) {
 	partyIDs := test.PartyIDs(N)
-	group := curve.Secp256k1{}
```

Comment:
@wydengyre these changes to `taprootDkg` seem misplaced in this PR. They should be functionally equivalent but given the fact that this is a forked repo and we'll need to sync with upstream I suggest we only make strictly necessary changes (even if it goes against linting rules, etc)